src/queue/manager.ts

```diff
@@ -0,0 +1,220 @@
+        if (newDocument) {
+            const addFileDriveJobs = jobs.filter(j => {
+                const actions = isOperationJob(j)
+                    ? j.operations
+                    : j.actions;
+
+                const op = actions.find((j: Action) => {
+                    const input = j.input as AddFileInput;
+                    return j.type === 'ADD_FILE' && input.id === job.documentId;
+                });
+
+                if (op) {
+                    return true;
+                }
+
+                return false;
+            })
+
+            score += 1;
+            addFileDriveJobs.forEach(j => {
+                dependencies.push(j.jobId);
+                // score += 1;
+            });
+        }
+
+        // if new job has add file operation then increase score of existing operations if existing arent already dependent on new job
+        const actions = isOperationJob(job) ? job.operations : job.actions;
+        const filteredActions = actions.filter((j: Action) => j.type === 'ADD_FILE');
+        for (const addFileOp of filteredActions) {
+            const input = addFileOp.input as AddFileInput;
+            const filteredJobs = jobs.filter(j => {
+                input.id === j.documentId;
+            }).map(j => {
+                if (j.dependencies && j.dependencies.includes(jobId)) {
+                    return j;
+                } else {
+                    return {
+                        ...j,
+                        score: j.score + 1,
+                        dependencies: [...j.dependencies ?? [], jobId]
+                    };
+                }
+            });
+            if (filteredJobs.length > 0) {
+                await queue.createOrUpdateJobs(filteredJobs);
+            }
+        }
+
+        // if new job has delete_node operation then remove existing operations from queue
+        const removeFileOps = actions.filter(
+            (j: Action) => j.type === 'DELETE_NODE'
+        );
+        for (const removeFileOp of removeFileOps) {
+            const input = removeFileOp.input as DeleteNodeInput;
+
+            const filteredJobs = jobs.filter(j => {
+                input.id === j.documentId;
+            });
+
+            await queue.removeJobs(filteredJobs.map(j => j.jobId));
+        }
+
+        // add job to queue
+        await queue.addJob({ jobId, score, dependencies, ...job });
+        this.emit('jobAdded', { jobId, score, dependencies, ...job });
+        return jobId;
+    }
+
+
+
+
+    private async processNextJob() {
+        if (!this.delegate) {
+            throw new Error('No server delegate defined');
+        }
+
+
+        const queue = this.queue;
+        const nextJob = await queue.getNextJob();
+        if (!nextJob) {
+            return;
+        }
+
+        console.log("next: ", nextJob)
+
+        try {
+            this.activeWorkers += 1;
+            const result = await this.delegate.processJob(nextJob);
```

Comment:
added base implementation of workers. does not work yet, but if you want feel free to look at  `queue/worker/node-cluster.ts` for an example how it could work